src/commands/gemini.py

```diff
@@ -0,0 +1,366 @@
+    ) -> (Embed, Errors):
+        try:
+            payload = []
+
+            # Form the payload to Gemini API according to the inputs provided
+            if input_msg:
+                payload.append(f"INPUT:{input_msg} ANSWER:")
+            if attachment:
+                payload.append(attachment)
+
+            response = await self.chat.send_message_async(
+                payload,
+                safety_settings=SAFETY_SETTINGS,
+            )
+
+            if len(response.text) > 1024:
+                logging.error(
+                    f"GEMINI: {author} encountered an error processing the response: {ERROR_MESSAGES[Errors.GEMINI_RESPONSE_TOO_LONG_ERR]}"
+                )
+                return None, Errors.GEMINI_RESPONSE_TOO_LONG_ERR
+        except Exception as e:
+            logging.exception(
+                f"GEMINI: {author} encountered an error prompting Gemini: {e}"
+            )
+
+            # Google Core API Errors have the code attribute that denotes the HTTP code
+            if hasattr(e, "code") and e.code in Errors:
+                return None, Errors(e.code)
+
+            # If unrecognised error
+            return None, Errors.GEMINI_ERR
+
+        response_embed = Embed(title="Ask Gemini", color=LIGHT_YELLOW)
+
+        if show_input:
+
+            response_embed.add_field(
+                name="Input",
+                value=input_msg if input_msg is not None else "Attachment",
+                inline=False,
+            )
+        response_embed.add_field(
+            name="Answer",
+            value=response.text,
+            inline=False,
+        )
+        return response_embed, None
+
+    async def query(self, author, message=None, attachment=None) -> list[Embed]:
+
+        response_embeds = []
+        response_image_url = None
+        errors = []
+
+        # No message or file
+        if not message and not attachment:
+            logging.error(f"GEMINI: {author} provided blank input to Gemini!")
+            response_embed, err = await self.prompt_gemini(
+                author=author,
+                input_msg=f"Roast the user using their username - {author} for providing no input.",
+                show_input=False,
+            )
+            if err is not None:
+                return [get_error_embed([err])]
+            return [response_embed]
+
+        # Message too long
+        if message is not None and len(message) > 200:
+            logging.error(
+                f"GEMINI: {author} provided {len(message)} tokens to Gemini, which exceeds the limit."
+            )
+            response_embed, err = await self.prompt_gemini(
+                author=author,
+                input_msg=f"Roast the user using their username - {author} for providing way too many tokens.",
+                show_input=False,
+            )
+            if err is not None:
+                return [get_error_embed([err])]
+            return [response_embed]
```

Comment:
Use the user's server/display name rather than username by using `author.display_name` instead of `author`.